src/Module/Api/BaseMastodon.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+
+namespace Friendica\Module\Api;
+
+use Friendica\Module\BaseApi;
+
+/**
+ * Mastodon API-specific methods
+ */
+class BaseMastodon extends BaseApi
+{
+	/**
+	 * Get current application from the Bearer token
+	 *
+	 * This Mastodon-specific Application is unrelated with the local Friendica application model.
+	 *
+	 * @return array token
+	 */
+	protected static function getCurrentApplication(): array
```

Comment:
Why is this moved into a separate class? Exactly this functionality will sooner or later be used in the Friendica part of the API as well, where we are currently using a different way to detect the application: https://github.com/friendica/friendica/blob/2021.06-rc/include/api.php#L116-L134

Reply:
Because so far this method is only used in Mastodon API calls, where "Application" seems to have a different meaning than the application model we have in Friendica with the `application`, `application-view`, and `application-token` table and view.

We do not control the terminology of the Mastodon API, but we do control it in our methods, hence the existence of this wrapper so that Mastodon methods make sense.

Reply:
I don't understand. So we have to define a second function that is doing the same when we intent to use it at the Friendica part of the API to fill the "app" field (which stands for "application") since on Mastodon it is named "Application" what we name "app"?

Reply:
What is the difference between the `getApplication` and `getCurrentApplication` methods in the current `BaseApi`? The latter simply returns the API token and yet this isn't what it's called. I've checked its uses and they all were related to what the Mastodon API calls "Application" which we can't change since we didn't define that term in that context.

Reply:
`getApplication` is used during the OAuth handling when generating an access token. It is not Mastodon specific but part of OAuth. It could make sense to isolate the OAuth specific stuff in a new class. Then we could also create a class for the "basic auth" handling that we are currently doing with our own API. And in the `BaseApi` class we could call the one or the other functionality when fetching the currently logged in user via a single function.

So we could have functions to return the currently logged in user and used application in the `BasicAuth` and the `OAuth` class. And these functions would only been called by two functions in the `BaseApi` class that would be called from everywhere.

Reply:
This is exactly the direction I wanted this PR to encourage.

Reply:
But this would make the `BaseMastodon` class obsolete :-)

Reply:
So be it, I'm not attached to it if we have another way of clearing the ambiguity.

Reply:
BTW: Since you agreed that a separation between `BasicAuth` and `OAuth` would make sense - which would make `BaseMastodon` obsolete, wouldn't it be good to perform this step in this PR? This would make many renamings obsolete as well.

Reply:
It would drastically reduce the number of modified files, but where should the method that currently is in the `BaseMastodon` class go?

Reply:
SHouldn't the `OAuth` part go to `Security\OAuth.php` and `BasicAuth` go to `Security\BasicAuth.php`?

Reply:
Sure, but it sounds like a larger changeset to create both of those.

Reply:
Not so large like renaming the base class in a lot of classes.

Reply:
See PR https://github.com/friendica/friendica/pull/10380